docs/dev/topology-over-raft.md

```diff
@@ -78,20 +78,20 @@ check if we need to rebalance. If so, it computes an incremental tablet migratio
 plan, persists it by moving tablets into transitional states, and moves the state machine
 into the tablet migration track. All this happens atomically form the perspective
 of group0 state machine.
-The tablet migration track excludes with other topology changes, so node operations
-will have to wait for the plan to finish before they can take over the state machine.
 
-The tablet balancing track migrates a small bunch of tablets, decided by the
-loaded balancer, and then moves back the state machine to the idle state.
-This gives other topology changes a chance to start, and if there aren't any, the
-load balancer will be called again to check the conditions. This way
-we can avoid blocking topology changes for too long, but also drive the cluster
-to eventually achieve balance in the absence of other requests.
+The tablet migration track also invokes the load balancer and starts new migrations
+to keep the cluster saturated with streaming. The load balancer is invoked
+on transition of tablet stages, and also continuously as long as it generates
+new migrations.
+
+If there is a pending topology change request, the load balancer
+will not be invoked to allow for current migrations to drain, after which the
+state machine will exit the tablet migration track and allow pending topology
+operation to start.
 
-The load balancer is always invoked with no pending tablet migrations. This
-allows for simplicity in the implementation, but may lead to underutilization
-of cluster resources if different tablets migrate with different speeds,
-and thus limit the speed of load balancing.
+The tablet migration track excludes with other topology changes, so node operations
+will have to wait for tablet migration track to finish before they can take over
+the state machine.
```

Comment:
Will node operations be prioritized? Where is this made explicit in the code?

Reply:
They are. Tablet load balancing is stopped when we notice pending node operation:

```

        // In order to keep the cluster saturated, ask the load balancer for more transitions.
        // Unless there is a pending topology change operation.
        auto [preempt, new_guard] = should_preempt_balancing(std::move(guard));
        guard = std::move(new_guard);
        if (!preempt) {
            auto plan = co_await balance_tablets(get_token_metadata_ptr());
            co_await generate_migration_updates(updates, guard, plan);
        }
```